src/WebJobs.Extensions.DurableTask/Listener/FunctionExecutionHelper.cs

```diff
@@ -0,0 +1,144 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host.Executors;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DurableTask.Listener
+{
+    internal static class FunctionExecutionHelper
+    {
+        public static async Task<WrappedFunctionResult> ExecuteFunctionInOrchestrationMiddleware(
```

Comment:
This method is currently not used in my approach. I figured if we know the app is shutting down, and the exception was thrown in the WebJobs namespace, that is sufficient to determine to abort the session. There may be false positives, but those will just be retried and fail when the host is not shutting down.

There is still theoretically some value in checking whether we got to the `executedUserCode` chunk, but at this point I don't think it is worth the code complexity.

Reply:
I would like it if we could keep the `executedUserCode` logic somehow. I'm not sure that we should tie it to the host shutdown, however, since there are other potential causes of user code not running (like DI failures).